batchflow/config.py

```diff
@@ -344,4 +289,66 @@ def items(self, flatten=False):
             items = self.flatten().items()
         else:
             items = super().items()
-        return items
\ No newline at end of file
+        return items
+
+    def __getitem__(self, key):
+        value = self._get(key)
+        return value
+
+    def __setitem__(self, key, value):
+        _ = self.pop(key, None)
+        self.put(key, value)
+
+    def __delitem__(self, key):
+        self.pop(key)
+
+    def __getattr__(self, key):
+        if key in self:
+            value = self.get(key)
+            value = Config(value) if isinstance(value, dict) else value
+            return value
+        raise AttributeError(key)
+    
+    def __iadd__(self, other):
+        if isinstance(other, dict):
+            self.update(other)
+        else:
+            raise TypeError(f"Unsupported operand type(s) for +=: 'Config' and '{type(other)}'")
+        return self
+
+    def __add__(self, other):
+        if isinstance(other, dict) and not isinstance(other, Config):
+            other = Config(other)
+        if isinstance(other, Config):
+            return Config([*self.flatten().items(), *other.flatten().items()])
+        return NotImplemented
+
+    def __radd__(self, other):
+        if isinstance(other, dict):
+            other = Config(other)
+        return other.__add__(self)
+
+    def __eq__(self, other):
+        self_ = self.flatten() if isinstance(self, Config) else self
+        other_ = Config(other).flatten() if isinstance(other, dict) else other
+        return self_.__eq__(other_)
```

Comment:
What if other is a `Config` and not `flatten`?

Reply:
> What if other is a `Config` and not `flatten`?

He will be flattened anyway because if other is an instance of `Config`, `isinstance(other, dict)` is True